cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -564,10 +569,12 @@ public synchronized void userTriggeredStopExecution() {
   /**
    * Request the executor to stop any ongoing execution.
    *
+   * @param forceExecutionStop Whether force execution to stop.
    * @return True if the flag to stop the execution is set after the call (i.e. was not set already), false otherwise.
    */
-  private synchronized boolean stopExecution() {
-    if (_stopRequested.compareAndSet(false, true)) {
+  private synchronized boolean stopExecution(boolean forceExecutionStop) {
```

Comment:
If an execution is already being stopped either due to a previous user request that is not a force stop or due to a dead broker during an ongoing execution (i.e. triggered by CC) -- then it is possible that the execution will be stuck during stopping an execution.

If the user sends a stop execution request with a force option enabled, we should allow cleaning up the relevant znodes (i.e. we should allow unclean stop requests while there is an ongoing clean stop).

Reply:
good catch!